test/core/parser/markers_test.py

```diff
@@ -1,43 +1,42 @@
-"""The Test file for The New Parser (Marker Classes)."""
-
-from sqlfluff.core.parser.markers import FilePositionMarker, EnrichedFilePositionMarker
-
-
-def test__markers__common_marker():
-    """Test construction and comparison of markers."""
-    # test making one from fresh
-    fp1 = FilePositionMarker()
-    fp2 = fp1.advance_by("abc")
-    fp3 = fp2.advance_by("def\nghi\njlk")
-    fp4 = fp3.advance_by("mno", idx=1)
-    # check comparisons
-    assert fp1 == FilePositionMarker(1, 1, 1, 0)
-    assert fp4 > fp3 > fp2 > fp1
-    assert fp1 < fp2 < fp3 < fp4
-    # Check advance works without newline
-    assert fp2 == FilePositionMarker(1, 1, 4, 3)
-    assert fp3 == FilePositionMarker(1, 3, 4, 14)
-    assert fp4 == FilePositionMarker(2, 3, 7, 17)
-
-
-def test__markers__common_marker_format():
-    """Test formatting of markers."""
-    fp1 = FilePositionMarker(1, 2, 3, 0)
-    # Check Formatting Style
-    assert str(fp1) == "[0](1, 2, 3)"
-
-
-def test__markers__enriched_marker_format():
-    """Test formatting of enriched markers."""
-    fp1 = EnrichedFilePositionMarker(
-        1,
-        1,
-        1,
-        0,
-        slice(0, 1),
-        slice(0, 1),
-        True,
-        source_pos_marker=FilePositionMarker(1, 2, 3, 0),
-    )
-    # Check Formatting Style
-    assert str(fp1) == "[0](1, 2, 3)"
+"""Tests for PositionMarker."""
+
+import pytest
+
+from sqlfluff.core.templaters import TemplatedFile
+from sqlfluff.core.parser.markers import PositionMarker
+
+
+@pytest.mark.parametrize(
+    "raw,start_pos,end_pos",
+    [
+        ("fsaljk", (0, 0), (0, 6)),
+        ("", (2, 2), (2, 2)),
+        ("\n", (2, 2), (3, 0)),
+        ("boo\n", (2, 2), (3, 0)),
+        ("boo\nfoo", (2, 2), (3, 3)),
+        ("\nfoo", (2, 2), (3, 3)),
+    ],
+)
+def test_markers__infer_next_position(raw, start_pos, end_pos):
+    """Test that we can correctly infer positions from strings."""
+    assert end_pos == PositionMarker.infer_next_position(raw, *start_pos)
+
+
+def test_markers__setting_position_raw():
+    """Test that we can correctly infer positions from strings & locations."""
+    templ = TemplatedFile.from_string("foobar")
+    # Check inference in the template
+    assert templ.get_line_pos_of_char_pos(2, True) == (1, 3)

```

Comment:
Nitpick: Could you pass the `True` / `False` values as keyword parameters? That'd make it a bit easier to understand what it means without having to refer to the called function.